mongoengine/queryset/manager.py

```diff
@@ -24,6 +27,7 @@ def __init__(self, queryset_func=None):
         if queryset_func:
             self.get_queryset = queryset_func
 
+    @retry_upon_reconnect
```

Comment:
Would this decorator be useful elsewhere? I mean, if there is a particular line, which throws the `AutoReconnect` exception, why not to inline this decorator around it?
The new public utility function has to be maintained. Also, I'd consider using this decorator outside of mongoengine a bad idea, because it would unnecessarily couple all reconnection retries in client's code to the one particular global(environment) variable.

But maybe, such an approach is idiomatic for mongoengine. @wojcikstefan, what do you think?